### PR TITLE
Enable to update env via App config

### PR DESCRIPTION
**Pull request by nakabonne**
**What this PR does / why we need it**:
Continued from https://github.com/pipe-cd/pipe/pull/3032#issuecomment-1012677731

Because an application config file has only env name, PipedAPI does have to fetch the env id by name. We don't want to add additional method to fetch by name as the Environment concept is deprecated, that's why I settled on using ListEnvironments for this case.

**Which issue(s) this PR fixes**:

Fixes https://github.com/pipe-cd/pipe/issues/3019

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/app/server/grpcapi/piped_api.go [hunk 1]
```diff
@@ -968,12 +968,46 @@ func (a *PipedAPI) UpdateApplicationConfigurations(ctx context.Context, req *pip
...
+		envMap[env.Name] = env.Id
+	}
+
 	for _, appInfo := range req.Applications {
+		envID := ""
+		if appInfo.EnvName != "" {
+			var ok bool
+			envID, ok = envMap[appInfo.EnvName]
+			if !ok {
+				a.logger.Warn(fmt.Sprintf("unknown environment name %q given", appInfo.EnvName))
```
Comment (nakabonne) [inline pkg/app/server/grpcapi/piped_api.go:1003]: Applied https://github.com/pipe-cd/pipe/pull/3032#discussion_r783734686

File: pkg/app/server/grpcapi/piped_api.go [hunk 2]
```diff
@@ -968,12 +968,46 @@ func (a *PipedAPI) UpdateApplicationConfigurations(ctx context.Context, req *pip
...
+		},
+	}
+	envs, err := a.environmentStore.ListEnvironments(ctx, opts)
+	if err != nil {
+		a.logger.Error("failed to get environments", zap.Error(err))
+		return nil, status.Error(codes.Internal, "Failed to get environments")
+	}
+	envMap := make(map[string]string, len(envs))
+	for _, env := range envs {
+		// It is assumed that the env name is kept unique by the user.
```
Comment (khanhtc1202) [inline pkg/app/server/grpcapi/piped_api.go:993]: nits, do we need this note?

**Comment by nakabonne** [comment]
Closing since we settled on disabling to update env completely.
/close

**Pull request closed without merging**